llvm/lib/Target/X86/X86ISelLowering.cpp

```diff
@@ -41125,6 +41255,154 @@ static SDValue combineShuffle(SDNode *N, SelectionDAG &DAG,
   return SDValue();
 }
 
+// Simplify a decomposed (sext (setcc)). Assumes prior check that
+// bitwidth(sext)==bitwidth(setcc operands).
+static SDValue simplifySExtOfDecomposedSetCCImpl(
```

Comment:
Any hope of pulling this out into its own patch?

Reply:
Will do

Reply:
@RKSimon seperate commit in this PR, or new PR as a whole?

Reply:
Can do, the only issue is that basically its functionality is dependent on this pr. Otherwise we never really get the necessary floating point flags to assist with the simplifcation.

Reply:
So this PR will have some regressions in it, ill post the other PR as rebased ontop of this one.

Reply:
Okay, dropped the simplifcations from this patch. See: https://github.com/llvm/llvm-project/pull/84360 (based on this one).

Reply:
Note, there are some regressions in this patch now but they should be cleared up in #84360.

Also note, the reason #84360 is second is so that we can properly test the `cmpp` code. Without this patch we never get the necessary FMF flags to actually do the `cmpp` simplifications. If you think that is unnecessary I can swap the order (#84360 does stand on its own mostly for is simplifications on `pcmpgt`).